packages/pluggableWidgets/datagrid-web/src/components/Table.tsx

```diff
@@ -82,10 +83,18 @@ export function Table<T>(props: TableProps<T>): ReactElement {
                 disableResizing: !column.resizable,
                 disableFilters: column.filterable === "no",
                 sortType: (rowA: Row<{ item: T }>, rowB: Row<{ item: T }>, columnId: IdType<object>): number => {
-                    const valueA = props.valueForFilterSort(rowA.values[columnId], Number(columnId)) || "";
-                    const valueB = props.valueForFilterSort(rowB.values[columnId], Number(columnId)) || "";
+                    const valueA = props.valueForSort(rowA.values[columnId], Number(columnId)) || "";
+                    const valueB = props.valueForSort(rowB.values[columnId], Number(columnId)) || "";
                     // Values should always be sorted in ASC mode https://github.com/tannerlinsley/react-table/pull/2504
-                    return valueA.localeCompare(valueB);
+                    if (typeof valueA === "string" && typeof valueB === "string") {
+                        return valueA.localeCompare(valueB);
+                    } else if (typeof valueA === "boolean" && typeof valueB === "boolean") {
+                        // True first
+                        return valueA === valueB ? 0 : valueA ? -1 : 1;
+                    } else if (valueA instanceof Date && valueB instanceof Date) {
+                        return (valueA as Date).getTime() - (valueB as Date).getTime();
+                    }
+                    return Number(valueA) - Number(valueB);
```

Comment:
Would there be a case where valueA would be string and valueB would be Date ?

Then this return would fail, so maybe we also exclusively check if valueA/B number and then cast to number ?

Reply:
Nope, but the typings are generic allowing 4 types.. so i also need to check type of B

valueA and B can be: string, boolean, Date, number or undefined. For the first 3 cases I am catching in the ifs, for the Big/Number and undefined it will be converted to Number.. `Number(undefined) = 0`